### PR TITLE
Add support for 50mW power level in link statistics

### PR DESCRIPTION
**Pull request by avsaase**
Requires https://github.com/ExpressLRS/ExpressLRS/pull/821 to actually send the TX power link statistic to the RX.

### CODE REVIEW

**Comment by CapnBry** [comment]
Come on. No one line PRs that can potentially cause issues if some system doesn't bounds check the value. There's a reason I didn't just slap it in there.

**Comment by avsaase** [comment]
Fair enough, but doesn't seem to cause problems in OpenTX. This raises the question I asked in the Discord, who can change the CRSF protocol?

**Comment by CapnBry** [comment]
TBS is the "owner" of the protocol but it is up to each project to make their own decisions about if they want to support it for us. TBS seemed agreeable to adding it but we have to get each project we want it in to take the PR to add it.

You just can't send an indexed value out that's outside the range of what literally every piece of software the speaks CRSF expects and just hope for the best. You need to at least get it supported by a majority of software before we start spitting it out.

**Comment by avsaase** [comment]
FYI, 50mW support was merged into OpenTX a week ago:
https://github.com/opentx/opentx/blob/ad64ce0ced3067517a81b9272d031c27bf631196/radio/src/telemetry/crossfire.cpp#L177

**Comment by CapnBry** [comment]
Awesome, sounds like you're on top of it. Betaflight looks like it has a bounds check as well, so it just needs to be added to uplinkTXPowerStatesMw, but also if you're making a PR there please also add 250mW and 50mW to `crsrRfPower_e` even though I do not think it is needed anywhere and shows a real attention to detail.
https://github.com/betaflight/betaflight/blob/master/src/main/telemetry/crsf.c#L259

It seems only iNav has the issue, unless Emuflight does as well. If you get it merged into these, I'll reopen this PR and take it.
* OpenTX
* EdgeTX
* iNav
* Betaflight

**Comment by avsaase** [comment]
Sweet. I'll get those PR's sorted this week.

**Comment by avsaase** [comment]
I opened the following PRs:

- https://github.com/betaflight/betaflight/pull/10930
- https://github.com/iNavFlight/inav/pull/7397
- https://github.com/EdgeTX/edgetx/pull/660
- https://github.com/emuflight/EmuFlight/pull/701

OpenTX support for 50 mW is included in 
- https://github.com/opentx/opentx/pull/8665

I also took a look at ArduPilot but all I could find was ([link](https://github.com/ArduPilot/ardupilot/blob/8035b92e6e346dc56959975be74672aad7374241/libraries/AP_RCProtocol/AP_RCProtocol_CRSF.h#L182)):
```
uint8_t uplink_tx_power; // ( enum 0mW = 0, 10mW, 25 mW, 100 mW, 500 mW, 1000 mW, 2000mW )
```

`uplink_tx_power` is not used anywhere and I can't find this enum so it seems AP doesn't do anything with the link power.

**Comment by CapnBry** [comment]
This is awesome and I'm going to go ahead and reopen this preemptively since you've done the legwork. Looks like the Betaflight patch you're trying to add index 8 = 25mW though 🙃

**Comment by avsaase** [comment]
@CapnBry typo on a tiny PR 🤦🏻‍♂️

**Comment by CapnBry** [comment]
> @CapnBry typo on a tiny PR 🤦🏻‍♂️

haha happens to the best of us. Glad it was you and not me, now attach the Stone of Shame!

**Review by JyeSmith — APPROVED** [review]

File: src/lib/POWERMGNT/POWERMGNT.cpp [hunk 1]
```diff
@@ -47,7 +47,7 @@ uint8_t POWERMGNT::powerToCrsfPower(PowerLevels_e Power)
     case PWR_500mW: return 4;
     case PWR_1000mW: return 5;
     case PWR_2000mW: return 6;
-    case PWR_50mW:
+    case PWR_50mW: return 8;
```
Comment (JyeSmith) [inline src/lib/POWERMGNT/POWERMGNT.cpp:50]: Minor, but do we want to order this by power level or returned value?

Comment (avsaase) [reply src/lib/POWERMGNT/POWERMGNT.cpp:50]: Imo this makes it more clear that the power levels are not in order.

Comment (avsaase) [reply src/lib/POWERMGNT/POWERMGNT.cpp:50]: @JyeSmith Do you want a new commit for this or should I amend it to the previous one? Some projects are picky about that and want to keep a clean commit history, others don't care.

Comment (JyeSmith) [reply src/lib/POWERMGNT/POWERMGNT.cpp:50]: I'm not picky, either is fine.

Comment (CapnBry) [reply src/lib/POWERMGNT/POWERMGNT.cpp]: Yeah we squash commit our PRs so even a new commit will get squashed down to 1

**Comment by avsaase** [comment]
Betaflight and Emuflight still need to merge my PRs but since they bounds check the uplink power this shouldn't cause any problems.

**Pull request merged**